app/helpers/login.ts

```diff
@@ -0,0 +1,19 @@
+import { Keypair } from "stellar-sdk";
+
+function isSecretKeyValid(secretKey: string): boolean {
+  try {
+    Keypair.fromSecret(secretKey);
+    return true;
+  } catch (error) {
+    return false;
+  }
+}
+
+export function login(secretKey: string): void {
+  if (isSecretKeyValid(secretKey)) {
+    const keypair: Keypair = Keypair.fromSecret(secretKey);
+    const publicKey = keypair.publicKey();
+    localStorage.setItem("publicKey", publicKey);
+    window.location.href = "/dashboard";
+  } else alert("Incorrect secret key");
+}
```

Comment:
This function is doing too much.

It says that logs in the user but it is retrieving the keypair from the secret and then the public key, storing that public key in the localstorage, routing the user to /dashboard and if the secret key is wrong it shows an message

Reply:
This function has been divided into smaller functions with more specific responsibilities.